packages/http-client-csharp/generator/Microsoft.Generator.CSharp/test/Providers/ModelTypeProviderTests.cs

```diff
@@ -170,5 +170,39 @@ public void BuildConstructor_ValidateConstructors()
             Assert.AreEqual(MethodSignatureModifiers.Public, initializationCtor.Signature.Modifiers);
             Assert.AreEqual(3, initializationCtor.Signature.Parameters.Count);
         }
+
+        [Test]
+        public void GenerateModelAsStruct()
+        {
+            var mockPluginInstance = new Mock<CodeModelPlugin>(_generatorContext) { };
+            var mockTypeFactory = new Mock<TypeFactory>() { };
+
+            var properties = new List<InputModelProperty>{
+                    new InputModelProperty("requiredString", "requiredString", "", InputPrimitiveType.String, true, false, false),
+                    new InputModelProperty("OptionalInt", "optionalInt", "", InputPrimitiveType.Int32, false, false, false),
+             };
+
+            mockTypeFactory.Setup(t => t.CreateCSharpType(It.IsAny<InputType>())).Returns((InputType inputType) =>
+            {
+                // Lookup the inputType in the list and return the corresponding CSharpType
+                var inputModelProperty = properties.Where(prop => prop.Type.Name == inputType.Name).FirstOrDefault();
+                if (inputModelProperty != null)
+                {
+                    return GetCSharpType(inputModelProperty.Type);
+                }
+                else
+                {
+                    throw new ArgumentException("Unsupported input type.");
+                }
+            });
+
+            mockPluginInstance.SetupGet(p => p.TypeFactory).Returns(mockTypeFactory.Object);
+            _mockPlugin?.SetValue(null, mockPluginInstance.Object);
+
+            var inputModel = new InputModelType("TestModel", "TestModel", "public", null, "Test model.", InputModelTypeUsage.RoundTrip, properties, null, Array.Empty<InputModelType>(), null, null, null, false, modelAsStruct: true);
+
+            var modelTypeProvider = new ModelProvider(inputModel);
+            Assert.AreEqual(TypeSignatureModifiers.Public | TypeSignatureModifiers.Struct | TypeSignatureModifiers.Partial | TypeSignatureModifiers.ReadOnly, modelTypeProvider.DeclarationModifiers);
```

Comment:
Can you add a test which creates the whole struct and asserts it against an expected string?